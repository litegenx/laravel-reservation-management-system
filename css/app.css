.router-link-active {
  cursor: pointer;
}

