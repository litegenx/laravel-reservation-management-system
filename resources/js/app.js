require('./bootstrap');
import 'babel-polyfill';
import Vue from 'vue';
import router from './router';
import store from './store';
import App from './components/App';
import vuetify from './plugins/vuetify';

new Vue({
    el: '#app',
    store,
    router,
    components: { App },
    template: '<App/>',
    vuetify,
});
