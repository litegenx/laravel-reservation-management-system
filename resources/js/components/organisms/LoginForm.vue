<template>
    <v-form @submit.prevent="login">
        <v-text-field
            v-model="loginForm.email"
            v-validate="'required|email'"
            :error-messages="errors.collect('email')"
            label="E-mail"
            data-vv-name="email"
            data-vv-validate-on="blur"
            required
        />
        <v-text-field
            v-model="loginForm.password"
            v-validate="'required|min:8'"
            :error-messages="errors.collect('password')"
            :append-icon="passwordIcon"
            :type="passwordType"
            name="input-10-1"
            label="Password"
            data-vv-name="password"
            data-vv-validate-on="blur"
            hint="8文字以上"
            counter
            required
            @click:append="passwordVisibility = !passwordVisibility"
        />
        <v-btn
            class="mt-4"
            type="submit"
        >
            Login
        </v-btn>
    </v-form>
</template>

<script>
    import { mapGetters } from 'vuex';

    export default {
        data () {
            return {
                loginForm: {
                    email: '',
                    password: '',
                },
                passwordVisibility: false,
            };
        },
        computed: {
            ...mapGetters({
                isAuthenticated: 'auth/isAuthenticated',
            }),
            passwordIcon: function () {
                return this.passwordVisibility ? 'visibility' : 'visibility_off';
            },
            passwordType: function () {
                return this.passwordVisibility ? 'text' : 'password';
            },
        },
        methods: {
            async login () {
                this.$validator.validateAll().then(async result => {
                    if (!result) {
                        return;
                    }
                    await this.$store.dispatch('auth/login', this.loginForm);
                });
            },
        },
    };
</script>
