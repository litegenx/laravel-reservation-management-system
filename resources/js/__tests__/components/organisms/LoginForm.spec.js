import { mount } from '@vue/test-utils';
import Vuex from 'vuex';
import LoginForm from '../../../components/organisms/LoginForm';
import app from '../app';
import setupLocalVue from '../setupLocalVue';
import flushPromises from 'flush-promises';

describe('LoginForm', () => {
    it('should render login form', async () => {
        const login = jest.fn(() => setTimeout(() => {}, 100));
        const wrapper = mount(app(LoginForm, 'inner'), setupLocalVue({
            store: new Vuex.Store({
                state: {
                    loginForm: {
                        email: '',
                        password: '',
                    },
                    passwordVisibility: false,
                },
                actions: {
                    'auth/login': login,
                },
            }),
        }));
        expect(wrapper.element).toMatchSnapshot();

        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.findAll('input')).toHaveLength(2);
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(1);
        expect(wrapper.findAll('input[type="password"]')).toHaveLength(1);
        expect(wrapper.findAll('button')).toHaveLength(1);
        expect(wrapper.find('button').text()).toBe('Login');
        expect(wrapper.findAll('.v-input__icon--append i')).toHaveLength(1);

        wrapper.find('.v-input__icon--append i').trigger('click');
        await flushPromises();
        expect(wrapper.findAll('input[type="text"]')).toHaveLength(2);
        wrapper.find('.v-input__icon--append i').trigger('click');

        const email = wrapper.find('input[type="text"]');
        const password = wrapper.find('input[type="password"]');

        email.setValue('123');
        password.setValue('12345');
        expect(email.element.value).toBe('123');
        expect(password.element.value).toBe('12345');
        await flushPromises();
        expect(wrapper.find('button').is('[disabled]')).toBe(true);
        wrapper.find('button').trigger('click');
        await flushPromises();
        expect(wrapper.element).toMatchSnapshot();
        expect(wrapper.find('.v-counter').text()).toBe('5');
        expect(login).not.toBeCalled();

        email.setValue('test@example.com');
        password.setValue('password');
        expect(email.element.value).toBe('test@example.com');
        expect(password.element.value).toBe('password');
        await flushPromises();
        expect(wrapper.find('button').is('[disabled]')).toBe(false);
        wrapper.find('button').trigger('click');
        wrapper.find('button').trigger('click');
        await flushPromises();
        expect(wrapper.element).toMatchSnapshot();
        expect(wrapper.find('.v-counter').text()).toBe('8');
        expect(login).toBeCalledTimes(1);
    });
});
