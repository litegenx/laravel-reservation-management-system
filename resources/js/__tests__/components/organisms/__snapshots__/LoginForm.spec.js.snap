// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render login form 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <form
          class="text-center"
        >
          <span>
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-13"
                      style="left: 0px; position: absolute;"
                    >
                      E-Mail
                    </label>
                    <input
                      id="input-13"
                      required="required"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <span>
            <div
              class="v-input theme--light v-text-field"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      for="input-17"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="input-17"
                      required="required"
                      type="password"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        role="button"
                      >
                        visibility_off
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                  <div
                    class="v-counter theme--light"
                  >
                    0
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <button
            class="mt-4 v-btn v-btn--contained theme--light v-size--default"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
        Login
    
            </span>
          </button>
        </form>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`LoginForm should render login form 2`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      data-booted="true"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <form
          class="text-center"
        >
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted error--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light error--text"
                      for="input-13"
                      style="left: 0px; position: absolute;"
                    >
                      E-Mail
                    </label>
                    <input
                      id="input-13"
                      required="required"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light error--text"
                    role="alert"
                  >
                    <div
                      class="v-messages__wrapper"
                    >
                      <div
                        class="v-messages__message"
                      >
                        The E-Mail field must be a valid email
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted error--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light error--text"
                      for="input-17"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="input-17"
                      required="required"
                      type="password"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        class="v-icon notranslate v-icon--link material-icons theme--light error--text"
                        role="button"
                      >
                        visibility_off
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light error--text"
                    role="alert"
                  >
                    <div
                      class="v-messages__wrapper"
                    >
                      <div
                        class="v-messages__message"
                      >
                        The Password field must be at least 8 characters
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-counter theme--light"
                  >
                    5
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <button
            class="mt-4 v-btn v-btn--contained v-btn--disabled theme--light v-size--default"
            disabled="disabled"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
        Login
    
            </span>
          </button>
        </form>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`LoginForm should render login form 3`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <main
      app="true"
      class="v-content"
      classname="mb-5"
      data-booted="true"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <form
          class="text-center"
        >
          <span>
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-13"
                      style="left: 0px; position: absolute;"
                    >
                      E-Mail
                    </label>
                    <input
                      id="input-13"
                      required="required"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <span>
            <div
              class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light"
                      for="input-17"
                      style="left: 0px; position: absolute;"
                    >
                      Password
                    </label>
                    <input
                      id="input-17"
                      required="required"
                      type="password"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        class="v-icon notranslate v-icon--link material-icons theme--light"
                        role="button"
                      >
                        visibility_off
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <div
                      class="v-messages__wrapper"
                    />
                  </div>
                  <div
                    class="v-counter theme--light"
                  >
                    8
                  </div>
                </div>
              </div>
            </div>
          </span>
           
          <button
            class="mt-4 v-btn v-btn--contained theme--light v-size--default"
            type="submit"
          >
            <span
              class="v-btn__content"
            >
              
        Login
    
            </span>
          </button>
        </form>
        <div
          id="test-contents"
        >
          contents
        </div>
      </div>
    </main>
  </div>
</div>
`;
